src/lib/services/event-service.ts

```diff
@@ -81,15 +82,61 @@ export default class EventService {
         return events;
     }
 
+    isAPI(user: IUser | IApiUser): boolean {
+        return (user as IApiUser)?.type === ApiTokenType.ADMIN;
```

Comment:
This will return false for frontend and backend tokens, which I expect is intentional, but I wonder if we could reflect that with the name perhaps